Lombiq.Tests.UI/Services/OrchardCoreInstance.cs

```diff
@@ -136,11 +134,20 @@ public IEnumerable<IApplicationLog> GetLogs(CancellationToken cancellationToken
                 {
                     Name = Path.GetFileName(filePath),
                     FullName = Path.GetFullPath(filePath),
-                    ContentLoader = () => File.ReadAllTextAsync(filePath, cancellationToken),
+                    ContentLoader = () => GetFileContentAsync(filePath),
                 })
             : Enumerable.Empty<IApplicationLog>();
     }
 
+    // This needs to be public and instance-level. If it's private static then it causes the app under test not to
+    // launch under Ubuntu for some reason, while working under Windows.
```

Comment:
There are actually browser log errors, I downloaded the failure dumps and every test and attempt has the following log:
`[Severe] https://localhost:9066/ - Failed to load resource: the server responded with a status of 500 ()`
Based on this, do you have any suggestions on what should I look into a bit more? Tbh. I'm a bit lost with this one.

Reply:
That's a server-side error, just shows up in the browser log due to a non-2xx response code (accessibility checking failing is a similar red herring). These should also have corresponding Orchard log entries, which they don't. So, my hunch is that something is wrong with the log configuration as well.

I don't have a ready solution either. However, try to temporarily go back to the _NLog.config_ and `OrchardCoreInstance` version you used before (which we know is not suitable in the long run, with the locking log read and all), and let's see if there will be any logs then. If still not, then you can try increasing `LogLevel` to `Information` in _appsettings.json_, and see if the resulting logs contain any hints (if there are still no logs in the failure dump, then something is really messed up with logging; though it shouldn't be completely wrong, since `ErrorHandlingTests` pass). Naturally, try these out locally first before you commit.

Reply:
See https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2545614855/Writing+GitHub+Actions+workflows+and+actions#Temporary-configuration-during-troubleshooting-for-faster-builds for further tips.

Reply:
You can also use the workflow from the `issue/OSOE-433` branch with the `test-filter` parameter for easier test filtering.

Reply:
I skipped the majority of the tests already, but I will check out the branch later, thanks.
Also, I won't be able to see any logs until [OSOE-466](https://lombiq.atlassian.net/browse/OSOE-466) is done, correct?

Reply:
You will, just it's a bit less convenient. They are still all there, but all at once for all tests, in a _test.out_ file in the dump, or in the workflow output.

Reply:
Some information, just keep track of them:
There were 0 occurrences where both builds failed all the tests, so every time either Windows or Linux passes.
Couldn't repro this with fewer tests, both builds passed 6/6 times.
Couldn't repro this locally.
Finally, I managed to log at "info" level, and the log files are generated, although there are no errors in them.
The logs indicate that the features are being downloaded and enabled, and the assets on the homepage are being initialized and getting 200 status codes.
The last two messages in the logs maybe can indicate something useful:
`21:42:30.5875||||OrchardCore.Modules.ModularBackgroundService|INFO|'ModularBackgroundService' is stopping. `
`21:42:30.5875||||OrchardCore.Environment.Shell.Distributed.DistributedShellHostedService|INFO|'DistributedShellHostedService' is stopping. `
Is this something that can guide us somewhere or this is normal?

Reply:
These two log messages mean that the shell is being shut down. Try to correlate the timing of these log entries with when the failing tests were started: If the log entries happen before, then they aren't happening during the current test (i.e. when we see the blank page) but previously, when the setup was run. The setup runs once for all tests (during the first test that reaches that point), then a snapshot is created from it, and that snapshot is used to start all the other tests' apps. The "Starting waiting for the setup operation." and "Finished waiting for the setup operation." messages in the log are about this, when tests wait for the setup to finish (all tests are blocked waiting for the first test to finish running the setup). The single test actually running the setup will also have "Starting setup operation." and "Finished setup operation." messages, in addition to log entries about interacting with the setup screen in between.

Reply:
Thank you for explaining it in great detail.
Unfortunately, based on the timestamps these logs aren't happening during the tests, these are the setup logs.
I will probably write a post about this on the dev channel, maybe someone will chime in, because the fact that I can't really progress with this bug and thus I'm blocking other issues is starting to really bother me.

Reply:
And what some after these log entries in the test log? I.e. what happens after this in the test itself? Try to add some debug messages with `ITestOutputHelper` in various places where the UI Testing Toolbox to try to pinpoint what's still reached and what isn't.

When you download the failure dump, it'll also contain the given app instances' folders, with the SQLite DB and all. If you try to run the app locally with that, does that work?

Reply:
Yes, the app is running with the generated DB file, everything seems fine with it, the only odd thing is that the home and post images are missing.
After some debugging, I could find that the second log message is not reached [here](https://github.com/Lombiq/UI-Testing-Toolbox/blob/17574a1c5cc52b3a29bf2e0d5de3bea03b838c88/Lombiq.Tests.UI/Services/UITestExecutionSession.cs#L482). I will try to look into this a bit more.

Reply:
This just means that there's an exception during `GoToRelativeUrlAsync()`, which is the first thing after waiting for the setup operation. This is the exception we see in the build output, the accessibility one, that indicates an empty page.